generator/config/stage/search.yaml

```diff
@@ -3,16 +3,93 @@ name: $search
 link: 'https://www.mongodb.com/docs/manual/reference/operator/aggregation/search/'
 type:
     - stage
-encode: single
+encode: object
 description: |
     Performs a full-text search of the field or fields in an Atlas collection.
     NOTE: $search is only available for MongoDB Atlas clusters, and is not available for self-managed deployments.
 arguments:
     -
-        name: search
+        name: operator
+        mergeObject: true
+        type:
+            - searchOperator
+        description: |
+            Operator to search with.  You can provide a specific operator or use
+            the compound operator to run a compound query with multiple operators.
+    -
+        name: index
+        optional: true
+        type:
+            - string
+        description: |
+            Name of the Atlas Search index to use. If omitted, defaults to "default".
+    -
+        name: highlight
+        optional: true
+        type:
+            # @todo support "highlight" type object
+            # https://www.mongodb.com/docs/atlas/atlas-search/highlighting/
+            - object
+        description: |
+            Specifies the highlighting options for displaying search terms in their original context.
+    -
+        name: concurrent
+        optional: true
+        type:
+            - bool
+        description: |
+            Parallelize search across segments on dedicated search nodes.
+            If you don't have separate search nodes on your cluster, 
+            Atlas Search ignores this flag. If omitted, defaults to false.
```

Comment:
Elsewhere, I recall you specifying `default: <value>` in the YAML configs. Should that be done here? Likewise for the `index` option above and some other params below with default values.

Reply:
The `default` option is only when we want a value to be set by default. This is only when a value is required, otherwise we rely on the default value server-side.